src/framework/shortcuts/view/shortcutsmodel.cpp

```diff
@@ -57,11 +58,32 @@ QVariant ShortcutsModel::data(const QModelIndex& index, int role) const
         return QString::fromStdString(action.code) + action.title.qTranslated() + action.description.qTranslated()
                + sequencesToNativeText(shortcut.sequences);
     }
+    case RoleSection:
+        return SectionName(shortcut);
+    case RoleSectionValue:
+        return SectionName(shortcut).toString() + sequencesToNativeText(shortcut.sequences);
+    case RoleSectionKey:
+        return SectionName(shortcut).toString() + this->action(shortcut.action).title.qTranslated();
     }
 
     return QVariant();
 }
 
+const QVariant ShortcutsModel::SectionName(const Shortcut& shortcut) const
+{
+    return this->getCategory(this->action(shortcut.action).category);
+}
+
+QStringList ShortcutsModel::getSections() const
```

Comment:
A few code style comments in this area:
- `SectionName` should start with a lowercase `s`
- I don't like unnecessary uses of `this->` because it decreases readability (makes a simple thing look complicated)
- `getSections` preferably without `get` prefix (and then also with lowercase `s`)

Reply:
Sure Ill incorporate this and above suggestions